WordPress/openverse-frontend
# Create a proof-of-concept for feature flags

## Fixes


Implements WordPress/openverse#186

## Description


This PR creates a PoC for working with feature flags inside the Openverse frontend. This code adds
- feature flag constants and types for TS
- a feature flag store for storing flag statuses and computing feature states
- middleware to init the store from cookies
- a makeshift preference page to update the cookies based on preferences and to demo the functionality (will be deleted before this is merged and replaced with a solid framework)

## Testing Instructions

0. Check out the PR and run the dev server.
1. Visit /preferences and see how the preferences are stored.
2. Play with the feature toggles.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
